frontend/src/store/course/reducer.ts

```diff
@@ -30,6 +32,9 @@ type State = {
   isMentorChoosingEnabled: boolean;
   mentors: UserDetailsResponseDto[];
   categories: CategoryGetAllItemResponseDto[];
+  menteesByCourseId: UserDetailsResponseDto[];
+  areMenteesFetched: boolean;
```

Comment:
```suggestion
  menteesByCourseDataStatus: DataStatus;
```
avoid booleans for the data status